nannyml_cloud_sdk/model_evaluation/model.py

```diff
@@ -0,0 +1,372 @@
+    mutation removeDataFromDataSource($input: DataSourceDeleteInput!) {
+        delete_data_from_data_source(input: $input) {
+            id
+        }
+    }
+""")
+
+
+class MetricConfiguration(TypedDict):
+    enabled: bool
+    rope_lower_bound: Optional[float]
+    rope_upper_bound: Optional[float]
+    hdi_width: Optional[float]
+
+
+class Model:
+
+    @classmethod
+    def list(
+            cls, name: Optional[str] = None, problem_type: Optional[ProblemType] = None
+    ) -> List[ModelSummary]:
+        """List defined models.
+
+        Args:
+            name: Optional name filter.
+            problem_type: Optional problem type filter.
+
+        Returns:
+            List of models that match the provided filter criteria.
+        """
+        return execute(_LIST_MODELS, {
+            'filter': {
+                'name': name,
+                'problemType': problem_type,
+            }
+        })['evaluation_models']
+
+    @classmethod
+    def get(cls, model_id: str) -> ModelDetails:
+        """Get details for a model.
+
+        Args:
+            model_id: ID of the model to get details for.
+
+        Returns:
+            Detailed information about the model.
+        """
+        return execute(_READ_MODEL, {'id': int(model_id)})['evaluation_model']
+
+    @classmethod
+    def create(
+            cls,
+            schema: ModelSchema,
+            reference_data: pd.DataFrame,
+            hypothesis: HypothesisType,
+            classification_threshold: float,
+            metrics_configuration: Dict[PerformanceMetric, MetricConfiguration],
+            evaluation_data: Optional[pd.DataFrame] = None,
+            name: Optional[str] = None,
+            key_performance_metric: Optional[PerformanceMetric] = None,
+    ) -> ModelDetails:
+        """Create a new model.
+
+        Args:
+            schema: Schema of the model. Typically, created using
+                [Schema.from_df][nannyml_cloud_sdk.model_evaluation.Schema.from_df].
+            hypothesis: The type of hypothesis the model is trying to validate. This can be one of the following:
+                - `MODEL_PERFORMANCE_NO_WORSE_THAN_REFERENCE`: The model's performance is not worse than the reference.
+                - `MODEL_PERFORMANCE_WITHIN_RANGE`: The model's performance is within a specified range.
+            classification_threshold: The threshold used to turn predicted probabilities into binary predictions.
+            reference_data: Reference data to use for the model.
+            evaluation_data: Analysis data to use for the model. If the data contains targets, targets must always be
+                provided together with analysis data.
+            metrics_configuration: Configuration for each metric to be used in the model.
+            name: Optional name for the model. If not provided, a name will be generated.
+            key_performance_metric: Optional key performance metric for the model. If not provided, no performance
```

Comment:
KPM is also not optional here. For better or worse we've made the field mandatory in the GraphQL API